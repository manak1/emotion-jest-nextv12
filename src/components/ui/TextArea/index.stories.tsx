import { ComponentStory, ComponentMeta } from "@storybook/react"

import Component from "./index"

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: "TextArea",
  component: Component,
  argTypes: {
    name: { control: "text", defaultValue: "id" },
    label: { control: "text", defaultValue: "ラベル" },
  },
} as ComponentMeta<typeof Component>

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Component> = (args) => {
  return <Component {...args} />
}

export const Default = Template.bind({})

export const Optional = Template.bind({})
Optional.args = {
  isOptional: true,
}
