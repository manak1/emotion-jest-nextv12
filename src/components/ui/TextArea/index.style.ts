import styled from "@emotion/styled"

export const Wrapper = styled.div({
  display: "flex",
  flexDirection: "column",
})

export const TextArea = styled.textarea(
  {
    outline: "none",
    border: "1px solid #d3d3d3",
    marginTop: "4px",
    padding: "8px",
    minHeight: "180px",
    borderRadius: "8px",
    resize: "none",
  },
  ({ theme }) => ({
    fontSize: theme.fontSizes.default,
    ...theme.focus,
  })
)
