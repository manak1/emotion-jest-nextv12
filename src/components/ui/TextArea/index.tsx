import React from "react"
import Label from "~/components/ui/Label"
import * as Styled from "./index.style"

type Props = React.ComponentProps<"textarea"> & {
  name: string
  label: string
  isOptional: boolean
}

const TextArea: React.VFC<Props> = (props) => {
  const { name, label, isOptional, ...rest } = props
  return (
    <Styled.Wrapper>
      <Styled.Box>
        <Label label={label} htmlFor={name} />
        {isOptional && <Styled.OptionalText>任意</Styled.OptionalText>}
      </Styled.Box>
      <Styled.TextArea id={name} {...rest} />
    </Styled.Wrapper>
  )
}

export default TextArea
