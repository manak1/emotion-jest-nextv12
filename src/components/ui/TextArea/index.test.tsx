import { userEvent } from "@storybook/testing-library"
import { render, screen } from "~/test/test-utils"
import TextArea from "./index"

describe("TextAreaのテスト", () => {
  it("snapshot", () => {
    const { asFragment } = render(
      <TextArea name="test" label="テキストエリア" />
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it("入力するとonChangeが呼ばれる", () => {
    const onChange = jest.fn()
    render(<TextArea name="test" label="テキストエリア" onChange={onChange} />)
    userEvent.type(screen.getByRole("textbox"), "テスト")
    expect(onChange).toBeCalledTimes(3)
  })
  it("任意の文字が表示されている", () => {
    const onChange = jest.fn()
    render(
      <TextArea
        name="test"
        label="テキストエリア"
        isOptional
        onChange={onChange}
      />
    )
    expect(screen.getByText("任意")).toBeInTheDocument()
  })
})
