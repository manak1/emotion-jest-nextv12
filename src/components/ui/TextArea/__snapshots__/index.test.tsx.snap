// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAreaのテスト snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-j7qwjs"
  >
    <div
      class="css-69i1ev"
    >
      <label
        class="css-z6ajj0"
        for="test"
      >
        テキストエリア
      </label>
    </div>
    <textarea
      class="css-1vw70bv"
      id="test"
    />
  </div>
</DocumentFragment>
`;
